src/main/java/uk/gov/companieshouse/itemgroupworkflowapi/controller/ItemGroupController.java

```diff
@@ -13,96 +13,116 @@
+                               ItemGroupsService itemGroupsService,
+                               ItemGroupsValidator itemGroupsValidator) {
         this.logger = logger;
         this.itemGroupsService = itemGroupsService;
         this.itemGroupsValidator = itemGroupsValidator;
     }
-
+    /**
+     * Create item group from ItemGroupData<p>
+     * POST RequestID logged<p>
+     * Validation performed on item group, on failure return BAD_REQUEST<p>
+     * Item group already exists check performed, if already exists return CONFLICT<p>
+     * Item group created, on success return CREATED, on error return BAD_REQUEST<p>
+     */
     @PostMapping("${uk.gov.companieshouse.itemgroupworkflowapi.createitemgroup}")
     public ResponseEntity<Object> createItemGroup(final @RequestHeader(REQUEST_ID_HEADER_NAME) String requestId,
                                                   final @RequestBody ItemGroupData itemGroupData) {
         logRequestId(requestId);
-        List<String> errors = itemGroupsValidator.validateCreateItemPayload(itemGroupData);
+        List<String> errors = itemGroupsValidator.validateCreateItemData(itemGroupData);
 
         if (!errors.isEmpty()) {
-            return buildValidationResponse(errors, itemGroupData);
+            return buildValidationResponse(requestId, errors);
         }
 
         try {
-            if (itemGroupsService.doesItemGroupExist(itemGroupData)) {
-                return buildItemAlreadyExistsResponse(itemGroupData);
-            }
+            if (itemGroupsService.doesItemGroupExist(itemGroupData))
+                return buildItemAlreadyExistsResponse(requestId, itemGroupData);
 
             final ItemGroup savedItem = itemGroupsService.createItemGroup(itemGroupData);
-            return buildCreateSuccessResponse(savedItem);
-
-        } catch (Exception ex) {
-            return buildErrorResponse(ex, itemGroupData);
+            return buildCreateSuccessResponse(requestId, savedItem);
+        }
+        catch (Exception ex) {
+            return buildErrorResponse(requestId, ex, itemGroupData);
         }
     }
 
     private void logRequestId(String requestId) {
-        Map<String, Object> logMap = logger.createLogMap();
-        logMap.put(REQUEST_ID_LOG_KEY, requestId);
-        logger.getLogger().info("create item group: request id", logMap);
-    }
+        DataMap dataMap = new DataMap.Builder()
+            .requestId(requestId)
+            .build();
 
-    private ResponseEntity<Object> buildCreateSuccessResponse(final ItemGroup savedItem) {
+        logger.getLogger().info(REQUEST_ID_PREFIX, dataMap.getLogMap());
+    }
+    /**
+     * @return ResponseEntity.status(CREATED)
+     */
+    private ResponseEntity<Object> buildCreateSuccessResponse(String requestId,
+                                                              final ItemGroup savedItem) {
         DataMap dataMap = new DataMap.Builder()
+            .xRequestId(requestId)
```

Comment:
Should we be using `xRequestId` or `requestId`?

Reply:
Sorry, I see my question may have been unclear here. I was really asking about whether we should be using the DataMap Builder `xRequestId()` or `requestId()` method there. They appear to be equivalent?